docs/source/glossary.md

```diff
@@ -0,0 +1,59 @@
+(glossary)=
+# PyTorch Glossary
+
+This glossary provides definitions for terms commonly used in PyTorch documentation.
```

Comment:
@albanD - can you help to review the definitions?

Reply:
FYI this file with the definitions was already there, I haven't added any new ones yet, I don't think.

Reply:
The glossary.md is 5 years old so very much not accurate. Left comments below to update.